pyprocessing/renderer/tkinter/window.py

```diff
@@ -119,13 +148,23 @@ def set_arc(self, x, y, width, height, start, stop, mode):
             start=start, extent=angle, style=mode,
             fill=self.pp.namespace.fill.hex,
             outline=self.pp.namespace.stroke.hex,
+            width=self.pp.namespace.thickness,
         )
         self.queued_actions.append(action)
 
     def set_point(self, x, y):
+        if self.pp.namespace.cap == 'round':
+            draw_function = 'create_rectangle'
+        elif self.pp.namespace.cap == 'projecting':
+            draw_function = 'create_oval'
+        else:
+            # Cap type square doesn't make a point
+            return
```

Comment:
for performance reasons, when a condition returns, put it as early as possible in the function. No need evaluating the whole condition if python can return early.

so `if self.pp.namespace.cap not in ('round', 'projecting'): return`

Also, the elif is superfluous, you can just have 2 consecutive if statements here (which are then again more performant)